manticore/platforms/evm.py

```diff
@@ -0,0 +1,1760 @@
+'''
+Solidity / Smart contract VM
+Implements the yellow paper: http://gavwood.com/paper.pdf
+Get example contracts from here:
+https://ethereum.github.io/browser-solidity/#version=soljson-latest.js
+'''
+import random, copy
+from ..platforms.platform import *
+from ..core.smtlib import solver, TooManySolutions, Expression, Bool, BitVec, Array, Operators, Constant, BitVecConstant, ConstraintSet
+from ..core.state import ForkState, TerminateState
+from ..utils.helpers import issymbolic
+from ..utils.event import Eventful
+from ..core.smtlib.visitors import pretty_print, arithmetic_simplifier, translate_to_smtlib
+from ..core.state import Concretize,TerminateState
+import logging
+import sys, hashlib
+if sys.version_info < (3, 6):
+    import sha3
+
+logger = logging.getLogger('PLATFORM')
+
+TT256 = 2 ** 256
+TT256M1 = 2 ** 256 - 1
+TT255 = 2 ** 255
+TOOHIGHMEM = 0x100
+
+def to_signed(i):
+    return i if i < TT255 else i - TT256
+
+def pack(i, size=32):
```

Comment:
This looks like it this takes an int and packs it into a 32 byte string, msb first. We should either update the name or document this in the docstring; the implementation isn't obvious.